storybooknative/src/Button/index.js

```diff
@@ -1,13 +1,181 @@
+      [TYPE_OPTIONS.INFO]: theme.orbit.colorTextButtonInfoBordered,
+      [TYPE_OPTIONS.SUCCESS]: theme.orbit.colorTextButtonSuccessBordered,
+      [TYPE_OPTIONS.WARNING]: theme.orbit.colorTextButtonWarningBordered,
+      [TYPE_OPTIONS.CRITICAL]: theme.orbit.colorTextButtonCriticalBordered,
+      [TYPE_OPTIONS.FACEBOOK]: theme.orbit.colorTextButtonFacebookBordered,
+      [TYPE_OPTIONS.GOOGLE]: theme.orbit.colorTextButtonGoogleBordered,
+    },
+  };
+  return tokens[name][type];
+};
+
+export const StyledButton = styled(View, props => {
+  const { bordered, theme, block, width, circled, disabled } = props;
+  return {
+    justifyContent: "center",
+    alignItems: "center",
+    height: getSizeToken(TOKENS.heightButton, props),
+    backgroundColor: bordered
+      ? theme.orbit.backgroundButtonBordered
+      : getTypeToken(TOKENS.backgroundButton, props),
+    width: block ? "100%" : width || "auto",
+    borderWidth: bordered ? "1px" : "0px",
+    borderColor: getTypeToken(TOKENS.borderColorButton, props),
+    borderRadius: circled
+      ? getSizeToken(TOKENS.heightButton, props)
+      : theme.orbit.borderRadiusNormal,
+    padding: 0,
+    paddingLeft: getSizeToken(TOKENS.paddingButton, props),
+    paddingRight: getSizeToken(TOKENS.paddingButton, props),
+    opacity: disabled && theme.orbit.opacityButtonDisabled,
+  };
+});
+
+StyledButton.defaultProps = {
+  theme: defaultTokens,
+};
+
+const StyledText = styled(Text, props => {
+  const { bordered, theme } = props;
+  return {
+    color: bordered
+      ? getTypeToken(TOKENS.colorTextButtonBordered, props)
+      : getTypeToken(TOKENS.colorTextButton, props),
+    fontWeight: theme.orbit.fontWeightBold,
+    fontSize: getSizeToken(TOKENS.fontSizeButton, props),
+  };
+});
+
+StyledText.defaultProps = {
+  theme: defaultTokens,
+};
+
+const StyledTouchableWrapper = styled(View, props => ({
+  overflow: "hidden",
+  width: props.block ? "100%" : props.width || "auto",
+  borderRadius: props.circled
+    ? getSizeToken(TOKENS.heightButton, props)
+    : props.theme.orbit.borderRadiusNormal,
+}));
+
+StyledTouchableWrapper.defaultProps = {
+  theme: defaultTokens,
+};
+
+const Button = (props: Props) => {
+  const {
+    onPress,
+    children,
+    size = SIZE_OPTIONS.NORMAL,
+    type = TYPE_OPTIONS.PRIMARY,
+    ...rest
+  } = props;
+  const onPressHandler = onPress || (() => {});
+  let width;
+  if (props.block) {
+    width = "100%";
+  } else {
+    width = props.width || "auto";
+  }
+
+  if (Platform.OS === "android") {
+    return (
+      <StyledTouchableWrapper
+        size={size}
+        block={props.block}
+        width={props.width}
+        circled={props.circled}
+      >
+        <Touchable
+          disabled={props.disabled || !props.onPress}
+          onPress={onPressHandler}
+          width={width}
+        >
+          <StyledButton size={size} type={type} {...rest}>
+            <StyledText size={size} type={type} bordered={props.bordered}>
```

Comment:
I am also wondering here if it would be possible to reuse the `Text` Orbit component.

Reply:
They don't use the same tokens for text color.